temporal-sdk/src/main/java/io/temporal/internal/replay/WorkflowExecutorCache.java

```diff
@@ -101,28 +114,50 @@ private WorkflowRunTaskHandler getForProcessing(String runId, Scope metricsScope
     }
   }
 
-  void markProcessingDone(String runId) {
+  void markProcessingDone(WorkflowExecution workflowExecution) {
     cacheLock.lock();
     try {
-      inProcessing.remove(runId);
+      inProcessing.remove(workflowExecution.getRunId());
+      log.debug(
+          "Workflow Execution {}-{} has been marked as not in-progress",
+          workflowExecution.getWorkflowId(),
+          workflowExecution.getRunId());
     } finally {
       cacheLock.unlock();
     }
   }
 
-  public void addToCache(String runId, WorkflowRunTaskHandler workflowRunTaskHandler) {
-    cache.put(runId, workflowRunTaskHandler);
+  public void addToCache(
+      WorkflowExecution workflowExecution, WorkflowRunTaskHandler workflowRunTaskHandler) {
+    cache.put(workflowExecution.getRunId(), workflowRunTaskHandler);
+    log.debug(
+        "Workflow Execution {}-{} has been added to cache",
+        workflowExecution.getWorkflowId(),
+        workflowExecution.getRunId());
   }
 
-  public boolean evictAnyNotInProcessing(WorkflowExecution execution, Scope metricsScope) {
+  public boolean evictAnyNotInProcessing(WorkflowExecution inFavorOfExecution, Scope metricsScope) {
     cacheLock.lock();
     try {
       this.metricsScope.gauge(MetricsType.STICKY_CACHE_SIZE).update(size());
+      if (log.isDebugEnabled()) {
+        long size = cache.size();
+        if (size == inProcessing.size()) {
+          log.debug(
+              "Workflow Execution cache and collection of in-process executions have the same size of {}",
+              size);
+        }
+      }
       for (String key : cache.asMap().keySet()) {
-        if (!key.equals(execution.getRunId()) && !inProcessing.contains(key)) {
+        if (!key.equals(inFavorOfExecution.getRunId()) && !inProcessing.contains(key)) {
           cache.invalidate(key);
           this.metricsScope.gauge(MetricsType.STICKY_CACHE_SIZE).update(size());
           metricsScope.counter(MetricsType.STICKY_CACHE_THREAD_FORCED_EVICTION).inc(1);
+          log.debug(
+              "Workflow Execution {}-{} caused eviction of Workflow Execution with runId {}",
+              inFavorOfExecution.getWorkflowId(),
+              inFavorOfExecution.getRunId(),
```

Comment:
Could we override `toString` or whatever is needed here so you can just pass the workflow execution as the only thing to interpolate ?

Reply:
No, it's a proto instance.